Create method to update account balance for custom accounts

CarterMcAlister opened the pull request:
Created a method that can be used to update the balance of custom accounts.
Account name in Personal Capital and new balance are accepted as parameters.

This can be used to write programs that scrape data from sources not supported by Personal Capital, and update the balance in Personal Capital.

Commit:
Create method to update account balance for custom accounts

JamaicanMoose commented on personalcapital/personalcapital.js:
```diff
@@ -207,6 +208,38 @@ class PersonalCapital {
+    }
+
+    return selectedAccount[0];
+  }
```
Make this more concise with something like this:
```js
async getAccountByName(accountName /*String*/) {
    const accounts = await this.getAccounts();
    const selectedAccount = accounts.find(account => account.name === accountName);
    if (!selectedAccount) {
      throw new AccountNotFoundError(`Account ${accountName} not found!`);
    }
    return selectedAccount;
  }
```

JamaicanMoose replied:
Forgot, AccountNotFoundError should be defined on line 35 as `class AccountNotFoundError extends Error {}` and exported in the module.exports.

JamaicanMoose commented on personalcapital/personalcapital.js:
```diff
@@ -207,6 +208,38 @@ class PersonalCapital {
+      balance: newBalance,
+      currentBalance: newBalance,
+      availableBalance: newBalance
+    };
```
Payloads should be defined in payload.endpoint as a function. Also ...account seems a bit extreme, does this actually require the whole account object or just the accountid?

JamaicanMoose reviewed (changes requested):
Commit works and good catch on the password argument but needs some cleanup before merging.

JamaicanMoose commented:
Password argument change integrated in patch version 2.0.1.

Commit:
Refactor getAccountByName,  and move updatebalance payload to payload.endpoint

CarterMcAlister commented:
@JamaicanMoose Thanks for the feedback, I have made your requested changes. 
Inspecting the `updatebalance` call in the network tab shows that the PC website makes the call by including the entire account object. I did try with only the `accountid`, but it does not update the balance.

JamaicanMoose commented:
@CarterMcAlister i did some inspection myself and confirmed that the website does use many of the parameters from the account object when updating some manual account types. I also noted that updating OTHER_ASSETS requires very few (however your code works when updating it presumably because the server ignores it). If you remove ownershipType: null (which as far as I can tell would override any existing ownershipType in the account object and make a note in the readme and function comment noting the asset type the payload was designed for (and that the code might work for other account types) I'm all good with approving it.

Commit:
Add methods to support Manual Investment Holdings

Commit:
Update README with new methods

CarterMcAlister commented:
@JamaicanMoose I have removed ownership types and added clarifying comments as to what asset types are supported by `updateBalance`. I have also added methods to support the `Manual Investment Holdings` asset.
